lib/ansible/modules/network/iosxr/iosxr_logging.py

```diff
@@ -16,318 +16,673 @@
 ---
 module: iosxr_logging
 version_added: "2.4"
-author: "Trishna Guha (@trishnaguha)"
-short_description: Manage logging on network devices
+author:
+    - "Trishna Guha (@trishnaguha)"
+    - "Kedar Kekan (@kedarX)"
+short_description: Configuration management of system logging services on network devices
 description:
-  - This module provides declarative management of logging
+  - This module provides declarative management configuration of system logging (syslog)
     on Cisco IOS XR devices.
-extends_documentation_fragment: iosxr
 notes:
-  - Tested against IOS XR 6.1.2
+  - Tested against IOS XRv 6.1.2
 options:
   dest:
     description:
-      - Destination of the logs.
-    choices: ['on', 'hostnameprefix', console', 'monitor', 'buffered']
+      - Destination for system logging (syslog) messages.
+    choices: ['host', 'console', 'monitor', 'buffered', 'file']
   name:
     description:
-      - If value of C(dest) is I(file) it indicates file-name,
-        for I(user) it indicates username and for I(host) indicates
-        the host name to be notified.
+      - If value of C(dest) is:
+        * I(file) name indicates file-name.
+        * I(host) name indicates the host-name or ip-address of syslog server.
+  vrf:
+    description:
+      - vrf name when syslog server is configured, C(dest) = C(host)
+    default: default
+    version_added: 2.5
   size:
     description:
-      - Size of buffer. The acceptable value is in range from 307200 to
-        125000000 bytes.
-    default: 307200
+      - * Size of buffer when C(dest) = C(buffered). The acceptable value is in the range I(307200 to 125000000 bytes). Default: 307200
+        * Size of file when C(dest) = C(file). The acceptable value is in the range I(1 to 2097152)KB. Default: 2 GB
```

Comment:
`Default: 2 GB` will cause issues, it's assumed to be `key: value`
You can quote the whole line